File: app/src/main/java/com/example/foodorderingapp/MainActivity.java
package com.example.foodorderingapp;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import androidx.navigation.NavController;
import androidx.navigation.fragment.NavHostFragment;
import androidx.navigation.ui.NavigationUI;
import com.example.foodorderingapp.databinding.ActivityMainBinding;
import com.google.android.material.bottomnavigation.BottomNavigationView;

public class MainActivity extends AppCompatActivity {
    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        // Set up the navigation controller
        NavHostFragment navHostFragment = (NavHostFragment) getSupportFragmentManager().findFragmentById(R.id.fragmentContainerView);
        NavController navController = navHostFragment.getNavController();

        BottomNavigationView bottomNav = findViewById(R.id.bottomNavigationView);
        NavigationUI.setupWithNavController(bottomNav, navController);

        // Set onClickListener for the notification icon
        binding.notificationIcon.setOnClickListener(v -> {
            Notification_bottom_Fragment bottomSheetDialog = new Notification_bottom_Fragment();
            bottomSheetDialog.show(getSupportFragmentManager(), "tag");
        });
    }
}

//package com.example.foodorderingapp;
//
//        import androidx.appcompat.app.AppCompatActivity;
//        import android.os.Bundle;
//        import androidx.navigation.NavController;
//        import androidx.navigation.findNavController;
//        import androidx.navigation.ui.NavigationUI;
//        import com.example.foodorderingapp.databinding.ActivityMainBinding;
//        import com.google.android.material.bottomnavigation.BottomNavigationView;

//public class MainActivity extends AppCompatActivity {
//    private ActivityMainBinding binding;
//
//    @Override
//    protected void onCreate(Bundle savedInstanceState) {
//        super.onCreate(savedInstanceState);
//        binding = ActivityMainBinding.inflate(getLayoutInflater());
//        setContentView(binding.getRoot());
//
//        // Set up the navigation controller
//        NavController navController = findNavController(R.id.fragmentContainerView);
//        BottomNavigationView bottomNav = findViewById(R.id.bottomNavigationView);
//        NavigationUI.setupWithNavController(bottomNav, navController);
//
//        // Set onClickListener for the notification icon
//        binding.notificationIcon.setOnClickListener(v -> {
//            Notification_bottom_Fragment bottomSheetDialog = new Notification_bottom_Fragment();
//            bottomSheetDialog.show(getSupportFragmentManager(), "tag");
//        });
//    }
//}

File: app/src/main/java/com/example/foodorderingapp/MainActivity.kt
package com.example.foodorderingapp

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.navigation.findNavController
import androidx.navigation.ui.setupWithNavController
import com.example.foodorderingapp.databinding.ActivityMainBinding
import com.example.foodorderingapp.databinding.NotificationItemBinding
import com.google.android.material.bottomnavigation.BottomNavigationView

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        var navController = findNavController(R.id.fragmentContainerView)
        var bottomNav = findViewById<BottomNavigationView>(R.id.bottomNavigationView)
        bottomNav.setupWithNavController(navController)

        binding.notificationIcon.setOnClickListener {
            val bottomSheetDialogue = Notification_bottom_Fragment()
            bottomSheetDialogue.show(supportFragmentManager, "tag")
        }
    }
}